libraries/chain/apply_context.cpp

```diff
@@ -363,39 +464,53 @@ void apply_context::schedule_deferred_transaction( const uint128_t& sender_id, a
                   "Replacing a deferred transaction is temporarily disabled." );
 
-      // TODO: The logic of the next line needs to be incorporated into the next hard fork.
-      // add_ram_usage( ptr->payer, -(config::billable_size_v<generated_transaction_object> + ptr->packed_trx.size()) );
+      uint64_t orig_trx_ram_bytes = config::billable_size_v<generated_transaction_object> + ptr->packed_trx.size();
+      if( replace_deferred_activated ) {
+         add_ram_usage( ptr->payer, -static_cast<int64_t>( orig_trx_ram_bytes ) );
+      } else {
+         control.add_to_ram_correction( ptr->payer, orig_trx_ram_bytes );
+      }
 
       db.modify<generated_transaction_object>( *ptr, [&]( auto& gtx ) {
-            gtx.sender      = receiver;
-            gtx.sender_id   = sender_id;
-            gtx.payer       = payer;
-            gtx.published   = control.pending_block_time();
-            gtx.delay_until = gtx.published + delay;
-            gtx.expiration  = gtx.delay_until + fc::seconds(control.get_global_properties().configuration.deferred_trx_expiration_window);
-
-            trx_size = gtx.set( trx );
-         });
+         if( replace_deferred_activated ) {
+            gtx.trx_id = trx.id();
+         }
+         gtx.sender      = receiver;
+         gtx.sender_id   = sender_id;
+         gtx.payer       = payer;
+         gtx.published   = control.pending_block_time();
+         gtx.delay_until = gtx.published + delay;
+         gtx.expiration  = gtx.delay_until + fc::seconds(control.get_global_properties().configuration.deferred_trx_expiration_window);
+
+         trx_size = gtx.set( trx );
+      } );
    } else {
       db.create<generated_transaction_object>( [&]( auto& gtx ) {
-            gtx.trx_id      = trx.id();
-            gtx.sender      = receiver;
-            gtx.sender_id   = sender_id;
-            gtx.payer       = payer;
-            gtx.published   = control.pending_block_time();
-            gtx.delay_until = gtx.published + delay;
-            gtx.expiration  = gtx.delay_until + fc::seconds(control.get_global_properties().configuration.deferred_trx_expiration_window);
-
-            trx_size = gtx.set( trx );
-         });
+         gtx.trx_id      = trx.id();
+         gtx.sender      = receiver;
+         gtx.sender_id   = sender_id;
+         gtx.payer       = payer;
+         gtx.published   = control.pending_block_time();
+         gtx.delay_until = gtx.published + delay;
+         gtx.expiration  = gtx.delay_until + fc::seconds(control.get_global_properties().configuration.deferred_trx_expiration_window);
+
+         trx_size = gtx.set( trx );
+      } );
    }
 
-   EOS_ASSERT( control.is_ram_billing_in_notify_allowed() || (receiver == act.account) || (receiver == payer) || privileged,
-               subjective_block_production_exception, "Cannot charge RAM to other accounts during notify." );
+   EOS_ASSERT( control.is_builtin_activated( builtin_protocol_feature_t::ram_restrictions )
```

Comment:
Since `control.is_builtin_activated( builtin_protocol_feature_t::ram_restrictions )` is always called, it can be called above and stored in a local variable to avoid calling it twice in this method.

Reply:
Resolved by https://github.com/EOSIO/eos/pull/7147/commits/10b8ae10d0c6f832f5e2d787f0f91295a1d2c2ae